test/scripts-test.js

```diff
@@ -1,7 +1,194 @@
+    // no past trips 
+    const travelerNumber = 2;
+    const currentTravelerData = addDataToCurrentTraveler(travelerNumber, travelers, trips, destinations);
+
+    const result = getImageURLsOfPastTrips(currentTravelerData);
+
+    // empty array
+    expect(result).to.be.an('array').that.is.empty;
+  });
+
+  it('should return an empty array when currentTravelerData is not valid or has no trips', function () {
+   
+    const currentTravelerData = null; 
+    
+    const result = getImageURLsOfPastTrips(currentTravelerData);
+
+    // empty array
+    expect(result).to.be.an('array').that.is.empty;
+  });
+});
+
+
+//////PENDING TRIPS////////////////
+
+describe('getImageURLsOfPendingTrips', function () {
+  it('should return an array of image URLs and destinations for pending trips when currentTravelerData is valid with pending trips', function () {
+
+    const travelerNumber = 3;
+    const currentTravelerData = addDataToCurrentTraveler(travelerNumber, travelers, trips, destinations);
+
+    const result = getImageURLsOfPendingTrips(currentTravelerData);
+
+    // an array
+    expect(result).to.be.an('array');
+
+   
+   
+    const expected = [
+      {
+        image:  "https://images.unsplash.com/photo-1524396309943-e03f5249f002?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1567&q=80",
+        destination: "Paris, France"
+      },
+    ];
+
+    expect(result).to.deep.equal(expected);
+  });
+
+  it('should return an empty array when currentTravelerData has no pending trips', function () {
+    
+    const travelerNumber = 2;
+    const currentTravelerData = addDataToCurrentTraveler(travelerNumber, travelers, trips, destinations);
+
+    const result = getImageURLsOfPendingTrips(currentTravelerData);
+
+    // empty array
+    expect(result).to.be.an('array').that.is.empty;
+  });
+
+  it('should return an empty array when currentTravelerData is not valid or has no trips', function () {
+    
+    const currentTravelerData = null; 
+
+    const result = getImageURLsOfPendingTrips(currentTravelerData);
+
+    // empty array
+    expect(result).to.be.an('array').that.is.empty;
+  });
+});
+
+
+
+/////////FUTURE TRIPS////////////////////////
+
+describe('getImageURLsOfFutureTrips', function () {
+  it('should return an array of image URLs and destinations for future trips when currentTravelerData is valid with future trips', function () {
+  
+    const travelerNumber = 4;
+    const currentTravelerData = addDataToCurrentTraveler(travelerNumber, travelers, trips, destinations);
+
+    const result = getImageURLsOfFutureTrips(currentTravelerData);
+
+    //  an array
+    expect(result).to.be.an('array');
+
+    
+    const expected = [
+      {
+        image: "https://images.unsplash.com/photo-1517821362941-f7f753200fef?ixlib=rb-1.2.1&auto=format&fit=crop&w=1952&q=80",
+        destination: "Marrakesh, Morocco"
+      },
+    ];
+
+    expect(result).to.deep.equal(expected);
+  });
```

Comment:
@butlertree You could make tests like this one leaner by skipping some of the variables. 

Here's an example of how I might do it:
```
it('should return an array of image URLs and destinations for future trips when currentTravelerData is valid with future trips', function () {
  const currentTravelerData = addDataToCurrentTraveler(4, travelers, trips, destinations);
  expect(getImageURLsOfFutureTrips(currentTravelerData)).to.deep.equal([
      {
        image: "https://images.unsplash.com/photo-1517821362941-f7f753200fef?ixlib=rb-1.2.1&auto=format&fit=crop&w=1952&q=80",
        destination: "Marrakesh, Morocco"
      }
    ]
  });
```
Also, it's totally fine to have a test to check that the result is an array, but it isn't really _necessary_ since the expected result IS an array. So, if it deep equals, it's kinda checking both things for you at once. More bang for your buck! 💰